src/protocol/organization.rs

```diff
@@ -0,0 +1,316 @@
+
+#[pymethods]
+impl OrganizationStatsReq {
+    #[new]
+    fn new() -> PyResult<Self> {
+        Ok(Self(organization_stats::Req))
+    }
+
+    fn __repr__(&self) -> PyResult<String> {
+        Ok(format!("{:?}", self.0))
+    }
+
+    fn __richcmp__(&self, other: Self, op: CompareOp) -> PyResult<bool> {
+        Ok(match op {
+            CompareOp::Eq => self.0 == other.0,
+            CompareOp::Ne => self.0 != other.0,
+            _ => return Err(PyNotImplementedError::new_err("")),
+        })
+    }
+
+    fn dump<'py>(&self, py: Python<'py>) -> PyResult<&'py PyBytes> {
+        Ok(PyBytes::new(
+            py,
+            &self.0.clone().dump().map_err(ProtocolError::new_err)?,
+        ))
+    }
+}
+
+#[pyclass]
+#[derive(Clone)]
+pub(crate) struct UsersPerProfileDetailItem(pub organization_stats::UsersPerProfileDetailItem);
+
+#[pymethods]
+impl UsersPerProfileDetailItem {
+    #[new]
+    fn new(profile: &PyAny, active: u64, revoked: u64) -> PyResult<Self> {
+        let profile = py_to_rs_user_profile(profile)?;
+        Ok(Self(organization_stats::UsersPerProfileDetailItem {
+            profile,
+            active,
+            revoked,
+        }))
+    }
+
+    fn __repr__(&self) -> PyResult<String> {
+        Ok(format!("{:?}", self.0))
+    }
+
+    fn __richcmp__(&self, other: Self, op: CompareOp) -> PyResult<bool> {
+        Ok(match op {
+            CompareOp::Eq => self.0 == other.0,
+            CompareOp::Ne => self.0 != other.0,
+            _ => return Err(PyNotImplementedError::new_err("")),
+        })
+    }
+
+    #[getter]
+    fn profile(&self) -> PyResult<PyObject> {
+        rs_to_py_user_profile(&self.0.profile)
+    }
+
+    #[getter]
+    fn active(&self) -> PyResult<u64> {
+        Ok(self.0.active)
+    }
+
+    #[getter]
+    fn revoked(&self) -> PyResult<u64> {
+        Ok(self.0.revoked)
+    }
+}
+
+gen_rep!(
+    organization_stats,
+    OrganizationStatsRep,
+    { .. },
+    [NotAllowed, reason: Reason],
+    [NotFound],
+);
+
+#[pyclass(extends=OrganizationStatsRep)]
+pub(crate) struct OrganizationStatsRepOk;
+
+#[pymethods]
+impl OrganizationStatsRepOk {
+    #[new]
+    fn new(
+        data_size: u64,
+        metadata_size: u64,
+        realms: u64,
+        users: u64,
+        active_users: u64,
+        users_per_profile_detail: Vec<UsersPerProfileDetailItem>,
+    ) -> PyResult<(Self, OrganizationStatsRep)> {
+        let users_per_profile_detail = users_per_profile_detail
+            .into_iter()
+            .map(|detail| detail.0)
+            .collect();
+        Ok((
+            Self,
+            OrganizationStatsRep(organization_stats::Rep::Ok {
+                data_size,
+                metadata_size,
+                realms,
+                users,
+                active_users,
+                users_per_profile_detail,
+            }),
+        ))
+    }
+
+    #[getter]
+    fn data_size(_self: PyRef<'_, Self>) -> PyResult<u64> {
```

Comment:
```suggestion
    fn data_size(self: PyRef<'_, Self>) -> PyResult<u64> {
```

Reply:
self is keyword

Reply:
you can remove the `_`

Reply:
I want something like `Pin` https://doc.rust-lang.org/std/pin/index.html#pinning-is-not-structural-for-field

Reply:
```shell
error[E0658]: `pyo3::PyRef<'_, organization::OrganizationStatsRepOk>` cannot be used as the type of `self` without the `arbitrary_self_types` feature                                                   
   --> src/protocol/organization.rs:135:24
    |
135 |     fn data_size(self: PyRef<'_, Self>) -> PyResult<u64> {
    |                        ^^^^^^^^^^^^^^^
    |
    = note: see issue #44874 <https://github.com/rust-lang/rust/issues/44874> for more information
    = help: consider changing to `self`, `&self`, `&mut self`, `self: Box<Self>`, `self: Rc<Self>`, `self: Arc<Self>`, or `self: Pin<P>` (where P is one of the previous types except `Self`)
```